applications/tari_validator_node/src/dan_node.rs

```diff
@@ -115,54 +122,64 @@ impl DanNode {
                     "Base node returned {} asset(s) to process",
                     assets.len()
                 );
-                for asset in assets {
-                    if tasks.contains_key(&asset.public_key) {
-                        debug!(
-                            target: LOG_TARGET,
-                            "Asset task already running for asset '{}'", asset.public_key
-                        );
-                        continue;
-                    }
-                    if let Some(allow_list) = &dan_config.assets_allow_list {
-                        if !allow_list.contains(&asset.public_key.to_hex()) {
-                            debug!(
-                                target: LOG_TARGET,
-                                "Asset '{}' is not allowlisted for processing ", asset.public_key
-                            );
-                            continue;
+                if let Some(allow_list) = &dan_config.assets_allow_list {
+                    assets.retain(|(asset, _)| allow_list.contains(&asset.public_key.to_hex()));
+                }
+                for (asset, mined_height) in assets.clone() {
+                    monitoring.add_if_unmonitored(asset.clone());
+                    monitoring.add_state(asset.public_key, mined_height, true);
+                }
+                let mut known_active_public_keys = assets.into_iter().map(|(asset, _)| asset.public_key);
+                let active_public_keys = monitoring
+                    .get_active_public_keys()
+                    .into_iter()
+                    .cloned()
+                    .collect::<Vec<PublicKey>>();
+                for public_key in active_public_keys {
+                    if !known_active_public_keys.contains(&public_key) {
```

Comment:
Rather use this and remove the `itertools` dependency
```suggestion
                    if !known_active_public_keys.any(|pk| pk == public_key) {
```